ddtrace/sampler.py

```diff
@@ -92,3 +103,251 @@ def set_sample_rate_by_service(self, rate_by_service):
 
 # Default key for service with no specific rate
 RateByServiceSampler._default_key = RateByServiceSampler._key()
+
+
+class DatadogSampler(BaseSampler):
+    """
+    """
+    __slots__ = ('rules', 'rate_limit')
+
+    DEFAULT_RATE_LIMIT = 100
+    NO_RATE_LIMIT = -1
+
+    def __init__(self, rules=None, rate_limit=None):
+        """
+        Constructor for DatadogSampler sampler
+
+        :param rules: List of :class:`SamplingRule` rules to apply to the root span of every trace, default no rules
+        :type rules: :obj:`list` of :class:`SamplingRule`
+        :param rate_limit: Global rate limit (traces per second) to apply to all traces regardless of the rules
+            applied to them, default 100 traces per second
+        :type rate_limit: :obj:`int`
+        """
+        # Ensure rules is a list
+        if not rules:
+            rules = []
+
+        # Validate that the rules is a list of SampleRules
+        for rule in rules:
+            if not isinstance(rule, SamplingRule):
+                raise TypeError('Rule {!r} must be a sub-class of type ddtrace.sampler.SamplingRules'.format(rule))
+        self.rules = rules
+
+        # Configure rate limiter
+        if rate_limit is None:
+            rate_limit = self.DEFAULT_RATE_LIMIT
+        self.limiter = RateLimiter(rate_limit)
+
+    def sample(self, span):
+        """
+        Decide whether the provided span should be sampled or not
+
+        The span provided should be the root span in the trace.
+
+        :param span: The root span of a trace
+        :type span: :class:`ddtrace.span.Span`
+        :returns: Whether the span was sampled or not
+        :rtype: :obj:`bool`
+        """
+        # If there are rules defined, then iterate through them and find one that wants to sample
```

Comment:
This doesn't align with the proposal of the RFC:

> Multiple rules can be provided, with the more granular rule applied first according to (service->name->resource).

Any reason to prefer the definition order? 
The RFC proposal was as such mostly to avoid sneakly ordering issue (a service rule ends up being before a resource rule, taking over silently).

Reply:
I think we should focus on the approach that is easiest to explain to customers.

While yes, there is a chance they can mess up the ordering and sampling rules are not applied how they want, but we can explain everything very simply:

> Sampling rules are applied in order where the first rule that matches is applied. Rules should be ordered from most specific to least specific.